lib/rewards/server/subscribers/rewards-subscriber.js

```diff
@@ -0,0 +1,190 @@
+const { sc, Logger } = require('@reldens/utils');
+const { Modifier } = require('@reldens/modifiers');
+const { Reward } = require('../models/reward');
+
+class RewardsSubscriber
+{
+    static async giveRewards(playerSchema, targetObject, eventsManager)
+    {
+        let rewards = sc.get(targetObject, 'rewards', false);
+
+        let winningRewards = this.getWinningRewards(rewards);
+        if (0 === winningRewards.length) {
+            return false;
+        }
+        let repositories = this.getRepositories(targetObject);
+
+        let itemRewards = [];
+        for (let winningReward of winningRewards) {
+            let giveRewardResults = await this.giveReward(winningReward, repositories, playerSchema, async (item) => {
+                itemRewards.push({
+                    item,
+                    reward: winningReward
+                });
+            });
+            if (0 < giveRewardResults.length) {
+                Logger.error('There were error(s) on the reward ' + winningReward.id +
+                    ' given to the player ' + playerSchema.id, giveRewardResults);
+            }
+        }
+
+        await this.emitEvent(eventsManager, itemRewards, playerSchema, targetObject);
+        await this.updateUniqueRewards(winningRewards, targetObject.dataServer);
+        return true;
+    }
+
+    static async emitEvent(eventsManager, itemRewards, playerSchema, targetObject)
+    {
+        if (0 === itemRewards.length) {
+            return;
+        }
+        await eventsManager.emit('reldens.itemsRewardsDropped', {
+            itemRewards,
+            player_id: playerSchema.player_id,
+            target_object: targetObject,
+            client: playerSchema?.skillsServer?.client?.client
+        });
+    }
+
+    static async giveReward(reward, repositories, playerSchema, callbackFn)
+    {
+        let results = [];
+        if (reward.isItemType()) {
+            let result = await this.applyItemReward(repositories.itemRepository, reward, playerSchema, callbackFn);
+            if (!result.isSuccess) {
+                results.push(result);
+            }
+        }
+
+        if (reward.isModifierType()) {
+            let result = await this.applyModifierReward(repositories.rewardsModifiersRepository, reward, playerSchema);
+            if (!result.isSuccess) {
+                results.push(result);
+            }
+        }
+
+        if (reward.hasExperienceSet()) {
+            let result = await this.applyExperienceReward(reward, playerSchema);
+            if (!result.isSuccess) {
+                results.push(result);
+            }
+        }
+        return results;
+    }
+
+    static getRewardResult(isSuccess, message = '')
+    {
+        let result = {
+            isSuccess: isSuccess
+        };
+        if (!isSuccess) {
+            result['message'] = message;
```

Comment:
if message has empty as default you can jsut remove this method and use `{isSuccess, message}` whatever this used